ner_v1/detectors/base_detector.py

```diff
@@ -72,10 +72,50 @@ def _set_language_processing_script(self):
             raise NotImplementedError('Please enable translation or extend language support'
                                       'for %s' % self._source_language_script)
 
-    def detect(self, message=None, structured_value=None, fallback_value=None, **kwargs):
+    def detect_bulk(self, messages=None, **kwargs):
         """
         Use detector to detect entities from text. It also translates query to language compatible to detector
 
+        Args:
+            messages (list of strings): list of natural text(s) on which detection logic is to be run.
+        Returns:
+            dict or None: dictionary containing entity_value, original_text and detection;
+                          entity_value is in itself a dict with its keys varying from entity to entity
+
+        Example:
+            1) Consider an example of restaurant detection from a message
+
+                messages = ['i want to order chinese from  mainland china and pizza from domminos']
+                output = detect(message=message)
+                print output
+                    >> [{'detection': 'message', 'original_text': 'mainland china', 'entity_value':
+                    {'value': u'Mainland China'}}, {'detection': 'message', 'original_text': 'domminos',
+                    'entity_value': {'value': u"Domino's Pizza"}}]
+        """
+        if self._source_language_script != self._target_language_script and self._translation_enabled:
```

Comment:
Add a if messages is None then messages = []. Otherwise there will be Exception on line 98